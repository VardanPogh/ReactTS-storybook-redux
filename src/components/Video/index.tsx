import React from 'react'
import ReactPlayer from 'react-player'

interface Props {
  url: string
  controls: boolean
  width?: string
  height?: string
  playsinline?: boolean
}

const Video: React.FC<Props> = ({
  url,
  controls,
  width,
  height,
  playsinline,
}) => {
  return (
    <ReactPlayer
      url={url}
      controls={controls}
      width={width}
      height={height}
      playsinline={playsinline}
    />
  )
}

export default Video
