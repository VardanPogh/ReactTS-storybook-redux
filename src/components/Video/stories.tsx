import React from 'react'
import { withKnobs, text, boolean } from '@storybook/addon-knobs'

import Video from '.'

export default {
  title: 'Video',
  component: Video,
  decorators: [withKnobs],
}

export const Default: React.FC = () => (
  <Video
    url={text('url', 'videos/BatNavExplainer.mp4')}
    controls={boolean('Controls', true)}
    width={text('width', '300px')}
    height={text('height', '300px')}
    playsinline={boolean('Controls', true)}
  />
)
