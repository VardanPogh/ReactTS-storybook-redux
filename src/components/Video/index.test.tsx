import React from 'react'
import { render } from '@testing-library/react'

import Video from '.'

describe('Video Test', () => {
  test('Load Video', () => {
    const { container } = render(
      <Video
        url="https://www.youtube.com/watch?v=EngW7tLk6R8"
        controls
        width="300px"
        height="300px"
        playsinline
      />
    )
    expect(container).toMatchSnapshot()
  })
})
