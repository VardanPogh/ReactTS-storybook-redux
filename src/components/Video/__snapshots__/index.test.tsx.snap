// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Test Load Video 1`] = `
<div>
  <div
    style="width: 300px; height: 300px;"
  />
</div>
`;
